common/v2/utils/uuidUtils.ts

```diff
@@ -0,0 +1,107 @@
+import sha1 from 'js-sha1';
```

Comment:
When we copy code from another repo let's:
- place it in the `vendor` directory and add a commented link to source file so we know where it comes from.
- also copy the spec files whenever possible.